src/runtime/device_interface.cpp

```diff
@@ -324,7 +324,7 @@ WEAK int halide_device_and_host_free(void *user_context, struct buffer_t *buf) {
             } else {
                 return 0;
             }
-        } else {
+        } else if (buf && buf->host) {
```

Comment:
i can change the else if to be 
   else if (buf->host) 
i am not familiar with this code to know if buf should have been non null and thus there is some issue elsewhere.